internal/docker/deployment.go

```diff
@@ -118,6 +118,32 @@ func (d *Deployment) RegisterUser(t *testing.T, hsName, localpart, password stri
 	return client
 }
 
+// LoginUser within a homeserver and return an authenticatedClient. Fails the test if the hsName is not found.
+func (d *Deployment) LoginUser(t *testing.T, hsName, localpart, password string) *client.CSAPI {
+	t.Helper()
+	dep, ok := d.HS[hsName]
+	if !ok {
+		t.Fatalf("Deployment.Client - HS name '%s' not found", hsName)
+		return nil
+	}
+	client := &client.CSAPI{
+		BaseURL:          dep.BaseURL,
+		Client:           client.NewLoggedClient(t, hsName, nil),
+		SyncUntilTimeout: 5 * time.Second,
+		Debug:            d.Deployer.debugLogging,
+	}
+	dep.CSAPIClients = append(dep.CSAPIClients, client)
+	userID, accessToken, deviceID := client.LoginUser(t, localpart, password)
+
+	// remember the token so subsequent calls to deployment.Client return the user
+	dep.AccessTokens[userID] = accessToken
```

Comment:
I wonder if we should leave the original access token in place. I can't convince myself that we should do one or the other, so I'm happy to leave the code as-is.

Reply:
Hmm, I think this might be a bit magic actually. It makes more sense to me if the original device is always referred to by `deployment.Client`, and you can create devices "on the side" if you like.

I'm partial to removing this assignment, and just calling this detail out in the method's docstring.